draft-ietf-quic-tls.md

```diff
@@ -778,6 +778,21 @@ the connection that is hosted on stream 1.  This sequence number is not visible
 to QUIC.
 
 
+## Receiving Protected Packets
+
+Once an endpoint successfully receives a packet with a given packet number, it
+MUST discard all packets with higher packet numbers if they cannot be
+successfully unprotected with either the same key, or - if there is a key update
+- the next packet protection key (see {{key-update}}).  Similarly, a packet that
```

Comment:
The problem there is that it's such a bizarre corner case that you wouldn't necessarily have code for it.  If the packets arrive in the other order, you will drop the higher packet number because you don't even try to decrypt it.  Maintaining the invariant that order of processing doesn't matter is hard here.  It's also the case that the obvious implementation of this doesn't necessarily couple the choice of advancing the keys with updating the last received packet number (I've implemented this for DTLS, and while the code is nearby, it's much easier not to entangle these concepts).

I think that I would rather say "MAY kill the connection".  That leaves the possibility open, and I don't have to explain how tricky it is to get this all lined up.